server/src/internalClusterTest/java/org/opensearch/index/codec/ZstdNotEnabledIT.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec;
+
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST)
+public class ZstdNotEnabledIT extends OpenSearchIntegTestCase {
+
+    public void testZStdCodecsWithoutPluginInstalled() {
+
+        internalCluster().startNode();
+        final String index = "test-index";
+
+        // creating index with zstd or zstd_no_dict should fail if custom-codecs plugin is not installed
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> createIndex(
+                index,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                    .put("index.codec", randomFrom("zstd", "zstd_no_dict"))
+                    .build()
+            )
+        );
```

Comment:
Probably not the best place for randomization and instead test both values:

```suggestion
        for (String codec : List.of("zstd", "zstd_no_dict")) {
            assertThrows(
                IllegalArgumentException.class,
                () -> createIndex(
                    index,
                    Settings.builder()
                        .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
                        .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
                        .put("index.codec", codec)
                        .build()
                )
            );
        }
```